src/main/java/uk/gov/companieshouse/overseasentitiesapi/interceptor/UserAuthenticationInterceptor.java

```diff
@@ -48,18 +52,32 @@ public boolean preHandle(@NonNull HttpServletRequest request, @NonNull HttpServl
             return true;
         }
 
+
         // TokenPermissions should have been set up in the request by TokenPermissionsInterceptor
         final var tokenPermissions = getTokenPermissions(request)
                 .orElseThrow(() -> new IllegalStateException("UserAuthenticationInterceptor - TokenPermissions object not present in request"));
 
+        boolean isRoeUpdateJourneyRequest = isRequestROEUpdate(request);
+
         // Check the user has the company_incorporation=create permission
         boolean hasCompanyIncorporationCreatePermission = tokenPermissions.hasPermission(Key.COMPANY_INCORPORATION, Value.CREATE);
 
+        // Check the use has the company_oe_annual_update=create permission for ROE Update
+        boolean hasCompanyOEAnnualUpdateCreatePermission = tokenPermissions.hasPermission(Key.COMPANY_OE_ANNUAL_UPDATE, Value.CREATE);
+
         var authInfoMap = new HashMap<String, Object>();
         authInfoMap.put(TRANSACTION_ID_KEY, transactionId);
         authInfoMap.put("request_method", request.getMethod());
+        authInfoMap.put("is_roe_update_journey_request", isRoeUpdateJourneyRequest);
+        authInfoMap.put("has_company_oe_annual_update_create_permission", hasCompanyOEAnnualUpdateCreatePermission);
         authInfoMap.put("has_company_incorporation_create_permission", hasCompanyIncorporationCreatePermission);
 
+        if (isRoeUpdateJourneyRequest && hasCompanyOEAnnualUpdateCreatePermission) {
+                ApiLogger.debugContext(reqId, "UserAuthenticationInterceptor authorised with company_oe_annual_update=create permission",
+                        authInfoMap);
+                return true;
+        }
+
         if (hasCompanyIncorporationCreatePermission) {
```

Comment:
Just falling into this code block is unsafe - if the user is on the Update journey they should absolutely have the `COMPANY_OE_ANNUAL_UPDATE` permission and if not, the method should fail-fast and return 'false'. I think if we introduce the flow control at line 61 above (see other comment), that should address this.